core/src/main/java/com/digitalpebble/storm/crawler/bolt/ParserBolt.java

```diff
@@ -201,22 +204,24 @@ public void execute(Tuple tuple) {
         }
 
         // parse
+        String text;
         try {
             tika.getParser().parse(bais, teeHandler, md, parseContext);
             text = textHandler.toString();
         } catch (Exception e) {
-            LOG.error("Exception while parsing {}", url, e.getMessage());
+            String errorMessage = "Exception while parsing " + url + ": "
+                    + e.getMessage();
+            LOG.error(errorMessage);
             eventCounter.scope(
                     "error_content_parsing_" + e.getClass().getSimpleName())
                     .incrBy(1);
-            // send to status stream in case another component wants to
-            // update its status
-            // TODO add the source of the error in the metadata
-            collector.emit(
-                    com.digitalpebble.storm.crawler.Constants.StatusStreamName,
-                    tuple, new Values(url, metadata, Status.ERROR));
+            // send to status stream in case another component wants to update
+            // its status
+            metadata.setValue(Constants.STATUS_ERROR_SOURCE, "content parsing");
+            metadata.setValue(Constants.STATUS_ERROR_MESSAGE, errorMessage);
+            collector.emit(StatusStreamName, tuple, new Values(url, metadata,
+                    Status.ERROR));
             collector.ack(tuple);
-            eventCounter.scope("parse exception").incrBy(1);
```

Comment:
I presume you removed this because we already emit a more specific counter above. Leaving this one would give a more global view, regardless of the type of error.